code/modules/surgery/tools.dm

```diff
@@ -93,7 +93,58 @@
 	attack_verb = list("pumps", "siphons")
 	tool_behaviour = TOOL_BLOODFILTER
 	toolspeed = 1
+	var/load_sound = 'sound/weapons/shotguninsert.ogg' //Monkestation edit start
+	var/list/obj/item/reagent_containers/glass/beakers = list()
+	var/max_beakers = 1
 
+/obj/item/blood_filter/handle_atom_del(atom/A)
+	. = ..()
+	if(A in beakers)
+		beakers.Remove(A)
+
+/obj/item/blood_filter/examine(mob/user)
+	. = ..()
+	if(beakers.len > 0)
+		. += "It has a [beakers[1].name] loaded."
+		if(beakers[1].reagents.total_volume)
+			. += "<span class='notice'>It has [beakers[1].reagents.total_volume] unit\s of reagents.</span>"
+		else
+			. += "<span class='danger'>It's empty.</span>"
+	else
+		. += "It has a receptacle that can hold a beaker."
+
```

Comment:
Why have it be a list if its hardcoded to only take from the first index?

Reply:
This was done sloppily, yes.  It was copied from the syringe gun code, which has an alternate which supports a magazine of several containers.  So as-is it's kind of in a half-between form where it's partly set up to allow for a hypothetical second filter that loads multiple containers.

Do you think I should add an advanced filter?  Or should I go make all of this hard coded for a single beaker?  Or should I just not bother until/unless it's a concern?

Reply:
Honestly, having a advanced version that could hold multiple would be neat. It could even work faster like the other upgraded tools. Best way I can think of going about it would be changing the code of the current code to not be hardcoded to use one beaker, and then make a child with that max beakers var set to a different value